aws/resource_aws_organizations_account.go

```diff
@@ -0,0 +1,228 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/organizations"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsOrganizationsAccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsOrganizationsAccountCreate,
+		Read:   resourceAwsOrganizationsAccountRead,
+		Delete: resourceAwsOrganizationsAccountDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"joined_method": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"joined_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				ForceNew:     true,
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 50),
+			},
+			"email": {
+				ForceNew:     true,
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAwsOrganizationsAccountEmail,
+			},
+			"iam_user_access_to_billing": {
+				ForceNew:     true,
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{organizations.IAMUserAccessToBillingAllow, organizations.IAMUserAccessToBillingDeny}, true),
+			},
+			"role_name": {
+				ForceNew:     true,
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateAwsOrganizationsAccountRoleName,
+			},
+		},
+	}
+}
+
+func resourceAwsOrganizationsAccountCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).organizationsconn
+
+	// Create the account
+	createOpts := &organizations.CreateAccountInput{
+		AccountName: aws.String(d.Get("name").(string)),
+		Email:       aws.String(d.Get("email").(string)),
+	}
+	if role, ok := d.GetOk("role_name"); ok {
+		createOpts.RoleName = aws.String(role.(string))
+	}
+
+	if iam_user, ok := d.GetOk("iam_user_access_to_billing"); ok {
+		createOpts.IamUserAccessToBilling = aws.String(iam_user.(string))
+	}
+
+	log.Printf("[DEBUG] Account create config: %#v", createOpts)
+
+	var err error
+	var resp *organizations.CreateAccountOutput
+	err = resource.Retry(4*time.Minute, func() *resource.RetryError {
```

Comment:
I don't remember 100% because this code is nearly a year old.  After looking at it for a little bit I think the issue was that there's no API action to check the status of `CreateOrganization`.  I didn't realize this could be a problem until I wrote this resource so I put the poll here.  Allowing `organizations.ErrCodeFinalizingOrganizationException` as a `RetryableError` seems to back that up.

Reply:
Awesome information! If that's the case, maybe we (more like I 😄 ) should setup the organization resource to wait until the organization is finalized before returning that its done on creation. I'll make a note to check that myself.

Reply:
If you create an issue and assign it to me, I will get it done.